### PR TITLE
[IMP] purchase: set date_planned of po lines as per date_planned of PO

### PR DESCRIPTION
**Pull request by anr-odoo**
task : https://www.odoo.com/web#id=2032417&action=333&active_id=966&model=project.task&view_type=form&menu_id=4720

pad : https://pad.odoo.com/p/r.fda3cacae6d8e949acc4f90f7f444fda

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase/models/purchase.py [hunk 1]
```diff
@@ -167,8 +155,17 @@ def name_get(self):
     def create(self, vals):
         if vals.get('name', 'New') == 'New':
             vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
+        if vals.get('date_planned'):
+            for line in vals.get('order_line'):
+                line[2]['date_planned'] = vals.get('date_planned')
```
Comment (sswapnesh) [inline addons/purchase/models/purchase.py]: Use `resolve_2many_commands`

Comment (sswapnesh) [inline addons/purchase/models/purchase.py]: Not sure but just a review based on https://github.com/odoo/odoo/pull/28921/files#diff-39d0613187d36f16c541734e80925e9dR267 and https://github.com/odoo/odoo/pull/35058/files#diff-f3712147f4b25231caa371bd83265dbeR1509
Cc: @pedrobaeza

File: addons/purchase/models/purchase.py [hunk 2]
```diff
@@ -169,6 +157,13 @@ def create(self, vals):
             vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
         return super(PurchaseOrder, self).create(vals)
 
+    def write(self, vals):
+        res = super(PurchaseOrder, self).write(vals)
+        if vals.get('date_planned'):
+            for line in self.order_line:
+                line.date_planned = self.date_planned
+        return res
```
Comment (sle-odoo) [inline addons/purchase/models/purchase.py]:
``` python
        res = super(PurchaseOrder, self).write(vals)
        if vals.get('date_planned'):
            self.order_line.write({'date_planned': vals['date_planned']})
        return res
```

File: addons/purchase/models/purchase.py [hunk 3]
```diff
@@ -181,7 +176,10 @@ def copy(self, default=None):
             seller = line.product_id._select_seller(
                 partner_id=line.partner_id, quantity=line.product_qty,
                 date=line.order_id.date_order and line.order_id.date_order.date(), uom_id=line.product_uom)
-            line.date_planned = line._get_date_planned(seller)
+            if new_po.date_planned:
+                line.date_planned = new_po.date_planned
```
Comment (sle-odoo) [inline addons/purchase/models/purchase.py]: there's no need to call _select_seller in this branch of the if

File: addons/purchase/models/purchase.py [hunk 4]
```diff
@@ -549,6 +543,10 @@ def create(self, values):
         if values.get('display_type', self.default_get(['display_type'])['display_type']):
             values.update(product_id=False, price_unit=0, product_uom_qty=0, product_uom=False, date_planned=False)
 
+        order_id = values.get('order_id')
+        order = self.env['purchase.order'].browse(order_id)
+        if order.date_planned and 'date_planned' not in values:
+            values['date_planned'] = order.date_planned
```
Comment (sle-odoo) [inline addons/purchase/models/purchase.py]:
``` python
if 'date_planned' not in values:
    order = self.env['purchase.order'].browse(order_id)
    if order.date_planned:
        values['date_planned'] = order.date_planned
```

Comment (anr-odoo) [reply addons/purchase/models/purchase.py]: @sle-odoo done.

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo r-

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by xmo-odoo** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo r+

**Pull request closed without merging**